catroid/src/main/java/org/catrobat/catroid/ui/SpriteActivity.java

```diff
@@ -459,6 +460,7 @@ private void addBackgroundFromUri(Uri uri) {
 			currentSprite.getLookList().add(look);
 			if (onNewLookListener != null) {
 				onNewLookListener.addItem(look);
+				refreshScriptFragment();
```

Comment:
why do you need this? it works perfectly fine for adding looks without it (see `addLookFromUri(Uri uri)`) why is this different?

sorry for being so annoying, but I just want to be sure that we do not start any new cargo cults by things like these.

Reply:
**you are right! it works fine. so I deleted this line.**
but I'm sure that there was a problem with updating the spinner of bricks `When background changes to` and `Set background`, cause I remember I discussed it with you °_°. 
otherwise I wouldn't touch this method.

Reply:
I know that we talked about it. I don't seem to remember the exact problem with the `WhenBackgroundChangesBrick` but I know that we need it for any bricks that have a spinner containing sprites (because adding a sprite is a 2 part process, where the sprite is only added after we return to the `SpriteActivity`.